File: mobile/lib/models/outlet.dart
class Outlet {
  final int id;
  final String name;
  final String code;
  final int? beatId;
  final int? territoryId;
  final String? ownerName;
  final String? mobile;
  final String? address;
  final String? channel;
  final double? gpsLat;
  final double? gpsLng;
  final String? photoUrl;
  final bool isIncomplete;
  final List<String> missingFields;

  Outlet({
    required this.id,
    required this.name,
    required this.code,
    this.beatId,
    this.territoryId,
    this.ownerName,
    this.mobile,
    this.address,
    this.channel,
    this.gpsLat,
    this.gpsLng,
    this.photoUrl,
    this.isIncomplete = false,
    this.missingFields = const [],
  });

  factory Outlet.fromJson(Map<String, dynamic> json) => Outlet(
    id: json['id'],
    name: json['name'] ?? '',
    code: json['code'] ?? '',
    beatId: json['beat_id'],
    territoryId: json['territory_id'],
    ownerName: json['owner_name'],
    mobile: json['mobile'],
    address: json['address'],
    channel: json['channel'],
    gpsLat: (json['gps_lat'] as num?)?.toDouble(),
    gpsLng: (json['gps_lng'] as num?)?.toDouble(),
    photoUrl: json['photo_url'],
    isIncomplete: json['is_incomplete'] == true,
    missingFields: json['missing_fields'] != null ? List<String>.from(json['missing_fields']) : [],
  );

  bool get hasGps => gpsLat != null && gpsLng != null;

  String get channelLabel {
    switch (channel) {
      case 'general_trade': return 'GT';
      case 'modern_trade': return 'MT';
      case 'pharmacy': return 'Pharmacy';
      case 'horeca': return 'HoReCa';
      case 'institutional': return 'Institutional';
      default: return channel ?? 'General';
    }
  }
}

class Beat {
  final int id;
  final String name;
  final String code;
  final String beatType;
  final String? l1PositionName;
  final String? assignedUserName;
  final int activeOutletCount;
  final List<Outlet> outlets;

  Beat({
    required this.id,
    required this.name,
    required this.code,
    required this.beatType,
    this.l1PositionName,
    this.assignedUserName,
    this.activeOutletCount = 0,
    required this.outlets,
  });

  factory Beat.fromJson(Map<String, dynamic> json) => Beat(
    id: json['id'],
    name: json['name'] ?? '',
    code: json['code'] ?? '',
    beatType: json['beat_type'] ?? 'GT',
    l1PositionName: json['l1_position_name'],
    assignedUserName: json['assigned_user_name'],
    activeOutletCount: json['active_outlet_count'] ?? 0,
    outlets: [],
  );
}

File: app/models/outlet.py
import enum

from sqlalchemy import Boolean, Column, DateTime, Enum as SAEnum, Float, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.models.base import Base


class OutletStatus(str, enum.Enum):
    active = "active"
    inactive = "inactive"


class ShopType(str, enum.Enum):
    kirana = "kirana"
    medical = "medical"
    general = "general"
    supermarket = "supermarket"
    hardware = "hardware"
    other = "other"


class ChannelType(str, enum.Enum):
    GT = "GT"   # General Trade
    MT = "MT"   # Modern Trade
    pharmacy = "pharmacy"
    horeca = "horeca"
    institutional = "institutional"
    other = "other"


class Outlet(Base):
    __tablename__ = "outlets"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False)
    code = Column(String(100), unique=True, nullable=True, index=True)
    owner_name = Column(String(255))
    mobile = Column(String(20), unique=True, nullable=True, index=True)
    address = Column(Text)
    pincode = Column(String(6), nullable=True)
    gstin = Column(String(15), nullable=True)
    channel = Column(SAEnum(ChannelType), nullable=True)
    shop_type = Column(SAEnum(ShopType), nullable=True)
    external_id = Column(String(100), nullable=True, index=True)  # Fieldassist mapping
    beat_id = Column(Integer, ForeignKey("beats.id"), nullable=True)
    territory_id = Column(Integer, ForeignKey("geographies.id"), nullable=True)
    gps_lat = Column(Float)
    gps_lng = Column(Float)
    photo_url = Column(Text, nullable=True)
    erp_id = Column(String(100))
    status = Column(SAEnum(OutletStatus), default=OutletStatus.active, nullable=False)
    is_active = Column(Boolean, default=True, nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    beat = relationship("Beat", back_populates="outlets")
    territory = relationship("Geography", foreign_keys=[territory_id])

    def status_badge_cls(self) -> str:
        mapping = {
            OutletStatus.active: "bg-emerald-900/50 text-emerald-300 ring-emerald-600/20",
            OutletStatus.inactive: "bg-gray-700 text-gray-400 ring-gray-500/20",
        }
        return mapping.get(self.status, "bg-gray-700 text-gray-400")

    def channel_display(self) -> str:
        if self.channel:
            return self.channel.value
        return "—"

    def shop_type_display(self) -> str:
        if self.shop_type:
            return self.shop_type.value.replace("_", " ").title()
        return "—"
